src/vs/server/node/webClientServer.ts

```diff
@@ -128,6 +139,38 @@ export class WebClientServer {
 		return serveFile(this._logService, req, res, filePath, headers);
 	}
 
+	/**
+	 * Handle extension resources
+	 */
+	private async _handleWebExtensionResource(req: http.IncomingMessage, res: http.ServerResponse, parsedUrl: url.UrlWithParsedQuery): Promise<void> {
+		// Strip `/web-extension-resource/` from the path
+		const normalizedPathname = decodeURIComponent(parsedUrl.pathname!); // support paths that are uri-encoded (e.g. spaces => %20)
+		const path = normalize(normalizedPathname.substr('/web-extension-resource/'.length));
+
+		const url = URI.parse(path).with({
+			scheme: this._productService.extensionsGallery?.resourceUrlTemplate ? URI.parse(this._productService.extensionsGallery.resourceUrlTemplate).scheme : 'https',
+			authority: path.substring(0, path.indexOf('/')),
+			path: path.substring(path.indexOf('/') + 1)
+		}).toString(true);
+
+		const headers: IHeaders = {};
+		for (const header of req.rawHeaders) {
+			if (req.headers[header]) {
+				headers[header] = req.headers[header]![0];
+			}
+		}
+
+		const context = await this._requestService.request({
+			type: 'GET',
+			url,
+			headers
+		}, CancellationToken.None);
+
+		res.writeHead(context.res.statusCode || 500, context.res.headers);
+		const buffer = await streamToBuffer(context.stream);
+		return res.end(buffer.buffer);
```

Comment:
use `context.stream.pipe(res)` instead to reduce memory pressure on the server? `streamToBuffer` implementation will read the whole file into memory which should be avoided.

Reply:
👍

Reply:
`context.stream` is our custom `VSBufferReadableStream` and is not supporting `pipe`.